### PR TITLE
Fix configuration filtering and slit width format

### PR DESCRIPTION
**Pull request by cquiroz**

### CODE REVIEW

File: model/shared/src/main/scala/explore/model/formats.scala [hunk 1]
```diff
@@ -68,6 +68,17 @@ trait formats:
   val formatArcsec: Format[String, Angle] =
     Format(_.parseIntOption.map(Angle.arcseconds.reverseGet(_)), Angle.arcseconds.get(_).toString)
 
+  val formatDecimalArcsec: Format[String, Angle] =
```
Comment (rpiaggio) [inline model/shared/src/main/scala/explore/model/formats.scala:71]: Should we test the `Format` laws?

**Review by rpiaggio — APPROVED** [review]
LGTM

File: model/shared/src/main/scala/explore/modes/SpectroscopyModesMatrix.scala [hunk 2]
```diff
@@ -401,7 +401,7 @@ case class SpectroscopyModesMatrix(matrix: List[SpectroscopyModeRow]) {
         wavelength.forall(w => w >= r.λmin.value && w <= r.λmax.value) &&
         resolution.forall(_ <= r.resolution) &&
         range.forall(_ <= r.λdelta.value) &&
-        slitWidth.forall(_.toMicroarcseconds <= r.slitLength.value.toMicroarcseconds) &&
+        slitWidth.forall(_.toMicroarcseconds <= r.slitWidth.value.toMicroarcseconds) &&
```
Comment (swalker2m) [inline model/shared/src/main/scala/explore/modes/SpectroscopyModesMatrix.scala:404]: When the slit width is specified, it must be less than the configuration option slit width value?  In other words, in the requirements the slit width entered is the maximum value that the user will accept?

File: model-tests/jvm/src/test/scala/explore/modes/ModesSuite.scala [hunk 3]
```diff
@@ -64,7 +64,7 @@ class ModesSuite extends CatsEffectSuite {
       )
       // .flatTap(_.traverse(IO.println))
       .map(_.length)
-      .assertEquals(85)
+      .assertEquals(48)
```
Comment (swalker2m) [inline model-tests/jvm/src/test/scala/explore/modes/ModesSuite.scala:67]: Maybe helpful to add a couple of tests to check that the results match.  For example, to check that all results have a slit width `<= 1`?

File: model/shared/src/main/scala/explore/model/formats.scala (hunk 1 above)
Comment (cquiroz) [reply model/shared/src/main/scala/explore/model/formats.scala:71]: I'll make a separate PR about this

File: model/shared/src/main/scala/explore/modes/SpectroscopyModesMatrix.scala (hunk 2 above)
Comment (andrewwstephens) [reply model/shared/src/main/scala/explore/modes/SpectroscopyModesMatrix.scala:404]: I think slit *length* is the desired parameter here:
https://github.com/gemini-hlsw/explore-help-docs/edit/main/configuration/focal_plane.md

**Pull request merged**